app/iOSFS.m

```diff
@@ -0,0 +1,449 @@
+//
+//  iOSFS.m
+//  iSH
+//
+//  Created by Noah Peeters on 26.10.19.
+//
+
+#import <Foundation/Foundation.h>
+#import <UIKit/UIKit.h>
+#include <sys/stat.h>
+#include "SceneDelegate.h"
+#include "iOSFS.h"
+#include "kernel/fs.h"
+#include "kernel/errno.h"
+#include "fs/path.h"
+
+const NSFileCoordinatorWritingOptions NSFileCoordinatorWritingForCreating = NSFileCoordinatorWritingForMerging;
+
+@interface DirectoryPickerDelegate : NSObject <UIDocumentPickerDelegate>
+
+@property NSURL *url;
+@property pthread_mutex_t *mutex;
```

Comment:
Locks aren't really intended for waiting for something to happen. Should use a condition variable. cond_t from util/sync.h is preferred because it allows the process to be interrupted by a signal, with e.g. ctrl-c.